src/util/configHelpers.js

```diff
@@ -539,6 +601,60 @@ const validListingFields = (listingFields, listingTypesInUse) => {
   }, []);
 };
 
+const validUserFields = (userFields, userTypesInUse) => {
+  const keys = userFields.map(d => d.key);
+  const scopeOptions = ['public', 'private', 'metadata'];
+  const validSchemaTypes = ['enum', 'multi-enum', 'text', 'long', 'boolean'];
+
+  return userFields.reduce((acc, data) => {
+    const schemaType = data.schemaType;
+
+    const validationData = Object.entries(data).reduce(
+      (acc, entry) => {
+        const [name, value] = entry;
+
+        // Validate each property
+        const [isValid, prop] =
+          name === 'key'
+            ? validKey(value, keys)
+            : name === 'label'
+            ? validLabel(value)
+            : name === 'scope'
+            ? validEnumString('scope', value, scopeOptions, 'public')
+            : name === 'schemaType'
+            ? validEnumString('schemaType', value, validSchemaTypes)
+            : name === 'enumOptions'
+            ? validSchemaOptions(value, schemaType)
+            : name === 'showConfig'
+            ? validUserShowConfig(value)
+            : name === 'saveConfig'
+            ? validUserSaveConfig(value)
+            : [true, value];
+
+        const hasFoundValid = !(acc.isValid === false || isValid === false);
+        // Let's warn about wrong data in listing extended data config
+        if (isValid === false) {
+          console.warn(`Unsupported user extended data configurations detected (${name}) in`, data);
+        }
+
+        return { config: { ...acc.config, ...prop }, isValid: hasFoundValid };
+      },
+      { config: {}, isValid: true }
+    );
+
+    if (validationData.isValid) {
+      const hasIncludeForListingTypes = validationData.config?.includeForListingTypes;
```

Comment:
This should be about user types not listing types.

---

P.S:
The idea has been that if config does not include any limitation to smaller set of (listing|user) types, then we automatically fill in all the active (user|listing) types. 
=> the end result is that the code can rely on the existence of includeFor(Listing|User)Types.

P.P.S
I've been considering if we should change this behavior so that there's a known value for it (like "ALL")
I.e. both versions would be allowed:
- `includeForListingTypes: ['a', 'b'],`
- `includeForListingTypes: 'all',`

Basically, because categories would also need this kind of behavior and there can be 
a) gazillion categories and 
b) categories can be nested (a flat array doesn't make that much sense with them).

Reply:
Would this then mean that
- the default user conf would still have this structure, and the restructured data would have a top-level includeForUserTypes array
```
userTypeConfig: {
      limitToUserTypeIds: true,
      userTypeIds: ['a', 'b', 'c'],
    },
```
- or is limitToUserTypeIds: true/false an obsolete attribute?

Reply:
@SariSaar Good point (as I said in the other thread). Forget what I said about includeForListingTypes, I'll swap it to a more verbose **listingTypeConfig** wrapper.

Reply:
This has now been updated!